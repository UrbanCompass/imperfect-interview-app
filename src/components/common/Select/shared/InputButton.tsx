import styled from '@emotion/styled';

const InputButton = styled.button`
  background: none;
  padding: 0;
  margin: 0;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 50%;
  right: 8px;
  transform: translate(-30%, -50%);
  svg.cx-icon {
    vertical-align: unset;
  }

  &:hover {
    cursor: pointer;
  }
`;
InputButton.displayName = 'InputButton';

export { InputButton };
