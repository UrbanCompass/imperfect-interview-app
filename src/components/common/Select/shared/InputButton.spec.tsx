import React from 'react';
import { mount } from 'enzyme';

import { InputButton } from './InputButton';

test('InputButton', (): void => {
  expect(mount(<InputButton>my content</InputButton>)).toMatchSnapshot();
});
