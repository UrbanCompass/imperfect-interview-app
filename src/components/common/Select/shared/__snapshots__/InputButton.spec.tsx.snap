// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton 1`] = `
.emotion-0 {
  background: none;
  padding: 0;
  margin: 0;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 50%;
  right: 8px;
  -webkit-transform: translate(-30%,-50%);
  -ms-transform: translate(-30%,-50%);
  transform: translate(-30%,-50%);
}

.emotion-0 svg.cx-icon {
  vertical-align: unset;
}

.emotion-0:hover {
  cursor: pointer;
}

<InputButton>
  <button
    className="emotion-0"
  >
    my content
  </button>
</InputButton>
`;
